src/pages/home.jsx

```diff
@@ -9,19 +9,46 @@ export const Home = () => {
   const [data, setData] = useState(null);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState(false);
+  const [pageNum, setPageNum] = useState(1);
   // This useEffect() for fetching data when the route load
   useEffect(() => {
     setLoading(true);
-    PICKLY.getAllPosts()
+    PICKLY.getAllPosts(pageNum)
       .then(({ data }) => {
+        // console.log(data);
         setData(data.data);
         setLoading(false);
       })
       .catch(err => {
         setLoading(false);
         setError(true);
       });
-  }, [token]);
+  }, [token, pageNum]);
+
+  const fetchNewPost = e => {
+    setLoading(true);
+    setPageNum(pageNum + 1);
+    PICKLY.getAllPosts(pageNum)
+      .then(({ data }) => {
+        setData(data.push(data.slice(data.data.length)));
+        setLoading(false);
+        console.log('updated Data', data);
+      })
+      .catch(err => {
+        setLoading(false);
+        setError(true);
+      });
+  };
+
+  document.addEventListener('scroll', () => {
+    // console.log('First', window.innerHeight + window.scrollY);
+    // console.log('Second', document.body.offsetHeight);
+    if (window.innerHeight + window.scrollY >= document.body.offsetHeight) {
```

Comment:
abstract this logic in a separate function call it something descriptive, I think this logic in the future should be much more smarter than a simple "end of the page reached", we should let the user wait until he reaches the end of the feed, we can get new posts when he has jus 5 posts left or something like this. this is why it is important to extract this logic into its own function so we can easily maintain, test and improve this logic.